### PR TITLE
Replace Edge with Relationship in new Kernel API

### PR DESCRIPTION
**Pull request by fickludd**
Rename Edge to Relationship in the Kernel API and NeoLE. 

I think this replacement seriously degrades the readability of the code in many cases. I acknowledge the value of consistent naming, but making this PR was just painful...

### CODE REVIEW

**Comment by tinwelint** [comment]
@fickludd Why do you think it degrades readability? Which specific cases are worst and why?

File: community/kernel-api/src/main/java/org/neo4j/impl/kernel/api/NeighbourCursor.java [hunk 1]
```diff
@@ -329,27 +329,27 @@ public final boolean hasProperties()
...
-    public final void edges( EdgeGroupCursor cursor )
+    public final void relationships( RelationshipGroupCursor cursor )
     {
-        neighbours.edges( cursor );
+        neighbours.relationships( cursor );
     }
 
     @Override
-    public void outgoingEdges( EdgeGroupCursor groups, EdgeTraversalCursor edges )
+    public void outgoingRelationships( RelationshipGroupCursor groups, RelationshipTraversalCursor relationships )
```
Comment (fickludd) [inline community/kernel-api/src/main/java/org/neo4j/impl/kernel/api/NeighbourCursor.java]: This takes me twice as long to read as before.

File: community/kernel-api/src/test/java/org/neo4j/impl/kernel/api/RuntimeITBase.java [hunk 2]
```diff
@@ -137,50 +137,50 @@ public void shouldExpandByEdgeGroup()
...
+        r.write().relationshipCreate( n3, label3, n1 );
+        r.write().relationshipCreate( n3, label2, n1 );
+        r.write().relationshipCreate( n3, label1, n1 );
+        r.write().relationshipCreate( n3, label1, n1 );
 
         // when
         NodeCursor nodeCursor = r.cursorFactory().allocateNodeCursor();
-        EdgeTraversalCursor edgeCursor = r.cursorFactory().allocateEdgeTraversalCursor();
-        EdgeGroupCursor groupCursor = r.cursorFactory().allocateEdgeGroupCursor();
+        RelationshipTraversalCursor relationshipCursor = r.cursorFactory().allocateRelationshipTraversalCursor();
```
Comment (fickludd) [inline community/kernel-api/src/test/java/org/neo4j/impl/kernel/api/RuntimeITBase.java]: This too.

File: enterprise/runtime/neole/src/main/java/org/neo4j/impl/store/prototype/neole/RelationshipGroupCursor.java [hunk 3]
```diff
@@ -187,25 +187,25 @@ public int loopCount()
...
         {
-            store.edges( nodeReference(), edgeReference, edge );
-            if ( !edge.next() )
+            store.relationships( nodeReference(), relationshipReference, relationship );
+            if ( !relationship.next() )
             {
                 return 0;
             }
-            return source ? (int) edge.sourcePrevEdgeReference() : (int) edge.targetPrevEdgeReference();
+            return source ? (int) relationship.sourcePrevRelationshipReference() : (int) relationship.targetPrevRelationshipReference();
```
Comment (fickludd) [inline enterprise/runtime/neole/src/main/java/org/neo4j/impl/store/prototype/neole/RelationshipGroupCursor.java]: Also harder.

File: enterprise/runtime/neole/src/test/java/org/neo4j/impl/store/prototype/neole/RelationshipScanCursorTest.java [hunk 4]
```diff
@@ -99,77 +99,77 @@ protected void create( GraphDatabaseService graphDb )
...
         }
     }
 
     @Test
-    public void shouldNotAccessDeletedEdge() throws Exception
+    public void shouldNotAccessDeletedRelationship() throws Exception
     {
         // given
-        try ( EdgeScanCursor edges = graph.allocateEdgeScanCursor() )
+        try ( RelationshipScanCursor relationships = graph.allocateRelationshipScanCursor() )
```
Comment (fickludd) [inline enterprise/runtime/neole/src/test/java/org/neo4j/impl/store/prototype/neole/RelationshipScanCursorTest.java]: And here.

File: enterprise/runtime/neole/src/test/java/org/neo4j/impl/store/prototype/neole/RelationshipTraversalCursorTest.java [hunk 5]
```diff
@@ -171,62 +171,62 @@ public void shouldTraverseEdgesOfGivenType() throws Exception
...
     @Test
-    public void shouldFollowSpecificEdge() throws Exception
+    public void shouldFollowSpecificRelationship() throws Exception
     {
         // given
         try ( NodeCursor node = graph.allocateNodeCursor();
-              EdgeGroupCursor group = graph.allocateEdgeGroupCursor();
-              EdgeTraversalCursor edge = graph.allocateEdgeTraversalCursor() )
+              RelationshipGroupCursor group = graph.allocateRelationshipGroupCursor();
+              RelationshipTraversalCursor relationship = graph.allocateRelationshipTraversalCursor() )
```
Comment (fickludd) [inline enterprise/runtime/neole/src/test/java/org/neo4j/impl/store/prototype/neole/RelationshipTraversalCursorTest.java]: Whatabout here? Yes.

**Review by fickludd — COMMENTED** [review]
@tinwelint Some select places that I thought were annoying.

**Comment by tinwelint** [comment]
For me it's almost the other way around... Edge disappears in the other words simply because my brain looks for Node and Relationship. It's a taste thing, but first and foremost it's a conformity thing, making the lives easier for most developers if terms are the same across the stack.

**Review by tinwelint — APPROVED** [review]

**Comment by fickludd** [comment]
Ok, I'm not passionate enough to fight for `Edge` if no-one agrees with me... let's merge this and continue with our lives.

**Pull request merged**